ci-utils/tests.sh

```diff
@@ -31,21 +30,37 @@ case "$1" in
     ;;
 esac
 
-while true;
+max_time=60 # Time in secondes
+total_dockercompose_services=4
+for ((cpt=1;cpt<=$max_time;cpt++));
 do
-  STATE=$(docker ps | grep ${CONTAINER})
-  if [ -n "${STATE}" ]; then break; fi
-  docker service ls
-  docker service ps --no-trunc ${DOCKER_SERVICE_NAME}
+  if [ $cpt == ${max_time} ];
+  then
+    echo "Could not start services after ${max_time} seconds"
+    docker service ls
+    for srv in $(docker service ls | grep 0/1 | awk '{print $1}');
+    do
+      docker service ps --no-trunc $srv
+      docker service logs $srv
+    done
+    exit 1
+  fi
+
+  nbps=$(docker service ls | grep 1/1 | awk '{print $4}' | wc -l)
+  if [ ${nbps} -eq ${total_dockercompose_services} ];
+  then
+    echo "All services up, starting tests"
+    break;
+  fi
+
+  echo "Waiting for all services (${nbps}/4) to start ($cpt secondes)."
```

Comment:
why do we need this logic?

Reply:
This logic handles all stack errors and gives the process status and logs of failing service. Before this, we only was checking api start. I had test running for a long time without results because of redis that was not able to launch.